sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_AzureEventSourceListenerTests.cs

```diff
@@ -0,0 +1,163 @@
+                Debug.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Name }");
+                Debug.WriteLine($"\tThe Event Hub was created at: { properties.CreatedOn }, in UTC.");
+                Debug.WriteLine($"\tThe following partitions are available: [{ string.Join(", ", properties.PartitionIds) }]");
+            }
+            finally
+            {
+                await producer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task TraceListener()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_TraceListener
+
+#if SNIPPET
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+#else
+            var connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            var eventHubName = scope.EventHubName;
+#endif
+
+            using AzureEventSourceListener traceListener = AzureEventSourceListener.CreateTraceLogger(EventLevel.LogAlways);
+
+            EventHubProducerClient producer = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                EventHubProperties properties = await producer.GetEventHubPropertiesAsync();
+
+                Debug.WriteLine("The Event Hub has the following properties:");
+                Debug.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Name }");
+                Debug.WriteLine($"\tThe Event Hub was created at: { properties.CreatedOn }, in UTC.");
+                Debug.WriteLine($"\tThe following partitions are available: [{ string.Join(", ", properties.PartitionIds) }]");
+            }
+            finally
+            {
+                await producer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task CustomListener()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_CustomListener
+
+#if SNIPPET
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+#else
+            var connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            var eventHubName = scope.EventHubName;
+#endif
+
+            using AzureEventSourceListener customListener = new AzureEventSourceListener((args, message) =>
+            {
+                if (args.EventSource.Name.StartsWith("Azure-Identity"))
```

Comment:
This part is great!   Could we also show a small demonstration of filtering the messages?   I'm thinking it would be helpful to see how to screen by the event identifier and level.   Maybe, filter the Azure.Identity section to only trace verbose messages?    

Assuming that we change to the event publishing scenario for log capture, maybe restrict the Event Hubs section to the following ids:

- Id:  3   (Event Publish Start)
- Id: 4 (Event Publish Complete)
- Id: 5 (Event Publish Error)

Reply:
Great suggestion! Thanks!